test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java

```diff
@@ -181,6 +184,86 @@ protected void doRun() throws Exception {
         }
     }
 
+    public void testCleanup() throws Exception {
```

Comment:
should we also add another test which does a full-cluster restart during a snapshot  or a snapshot delete? This should result in repo garbage as well, and will be more realistic in the garbage that's created.

Reply:
We wouldn't fully pass that test yet :( As discussed, we are still leaking `snap-${uuid}.dat` blobs even with this fix. So if we force a full restart we will eventually get into the state of leaving those files. This PR only cleans up stale indices.

That said: I have another branch that also deals with this case but it has this one (and a few other cleanup PRs I raised) as a dependency and we can look into that once we're done with the stale indices I guess? Then I'm all for testing full cluster restarts :)